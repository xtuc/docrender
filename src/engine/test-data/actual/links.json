{
  "title": "publish-bot test - links",
  "body": {
    "content": [
      {
        "endIndex": 1,
        "sectionBreak": {
          "sectionStyle": {
            "columnSeparatorStyle": "NONE",
            "contentDirection": "LEFT_TO_RIGHT",
            "sectionType": "CONTINUOUS"
          }
        }
      },
      {
        "startIndex": 1,
        "endIndex": 3,
        "paragraph": {
          "elements": [
            {
              "startIndex": 1,
              "endIndex": 2,
              "person": {
                "personId": "kix.jc51obuvxl7m",
                "textStyle": {},
                "personProperties": {
                  "name": "Sven Sauleau",
                  "email": "Sven.sauleau@gmail.com"
                }
              }
            },
            {
              "startIndex": 2,
              "endIndex": 3,
              "textRun": {
                "content": "\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      },
      {
        "startIndex": 3,
        "endIndex": 18,
        "paragraph": {
          "elements": [
            {
              "startIndex": 3,
              "endIndex": 17,
              "textRun": {
                "content": "cloudflare.com",
                "textStyle": {
                  "underline": true,
                  "foregroundColor": {
                    "color": {
                      "rgbColor": {
                        "red": 0.06666667,
                        "green": 0.33333334,
                        "blue": 0.8
                      }
                    }
                  },
                  "link": {
                    "url": "http://cloudflare.com"
                  }
                }
              }
            },
            {
              "startIndex": 17,
              "endIndex": 18,
              "textRun": {
                "content": ".\n",
                "textStyle": {}
              }
            }
          ],
          "paragraphStyle": {
            "namedStyleType": "NORMAL_TEXT",
            "direction": "LEFT_TO_RIGHT"
          }
        }
      }
    ]
  },
  "documentStyle": {
    "background": {
      "color": {}
    },
    "pageNumberStart": 1,
    "marginTop": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginBottom": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginRight": {
      "magnitude": 72,
      "unit": "PT"
    },
    "marginLeft": {
      "magnitude": 72,
      "unit": "PT"
    },
    "pageSize": {
      "height": {
        "magnitude": 841.8897637795277,
        "unit": "PT"
      },
      "width": {
        "magnitude": 595.2755905511812,
        "unit": "PT"
      }
    },
    "marginHeader": {
      "magnitude": 36,
      "unit": "PT"
    },
    "marginFooter": {
      "magnitude": 36,
      "unit": "PT"
    },
    "useCustomHeaderFooterMargins": true
  },
  "namedStyles": {
    "styles": [
      {
        "namedStyleType": "NORMAL_TEXT",
        "textStyle": {
          "bold": false,
          "italic": false,
          "underline": false,
          "strikethrough": false,
          "smallCaps": false,
          "backgroundColor": {},
          "foregroundColor": {
            "color": {
              "rgbColor": {}
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          },
          "baselineOffset": "NONE"
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "alignment": "START",
          "lineSpacing": 115,
          "direction": "LEFT_TO_RIGHT",
          "spacingMode": "COLLAPSE_LISTS",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "unit": "PT"
          },
          "borderBetween": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderTop": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderBottom": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderLeft": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "borderRight": {
            "color": {},
            "width": {
              "unit": "PT"
            },
            "padding": {
              "unit": "PT"
            },
            "dashStyle": "SOLID"
          },
          "indentFirstLine": {
            "unit": "PT"
          },
          "indentStart": {
            "unit": "PT"
          },
          "indentEnd": {
            "unit": "PT"
          },
          "keepLinesTogether": false,
          "keepWithNext": false,
          "avoidWidowAndOrphan": true,
          "shading": {
            "backgroundColor": {}
          },
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_1",
        "textStyle": {
          "fontSize": {
            "magnitude": 20,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 20,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_2",
        "textStyle": {
          "bold": false,
          "fontSize": {
            "magnitude": 16,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 18,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 6,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_3",
        "textStyle": {
          "bold": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.2627451,
                "green": 0.2627451,
                "blue": 0.2627451
              }
            }
          },
          "fontSize": {
            "magnitude": 14,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 16,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_4",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 12,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 14,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_5",
        "textStyle": {
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "HEADING_6",
        "textStyle": {
          "italic": true,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 11,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "magnitude": 12,
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 4,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "TITLE",
        "textStyle": {
          "fontSize": {
            "magnitude": 26,
            "unit": "PT"
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 3,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      },
      {
        "namedStyleType": "SUBTITLE",
        "textStyle": {
          "italic": false,
          "foregroundColor": {
            "color": {
              "rgbColor": {
                "red": 0.4,
                "green": 0.4,
                "blue": 0.4
              }
            }
          },
          "fontSize": {
            "magnitude": 15,
            "unit": "PT"
          },
          "weightedFontFamily": {
            "fontFamily": "Arial",
            "weight": 400
          }
        },
        "paragraphStyle": {
          "namedStyleType": "NORMAL_TEXT",
          "direction": "LEFT_TO_RIGHT",
          "spaceAbove": {
            "unit": "PT"
          },
          "spaceBelow": {
            "magnitude": 16,
            "unit": "PT"
          },
          "keepLinesTogether": true,
          "keepWithNext": true,
          "pageBreakBefore": false
        }
      }
    ]
  },
  "revisionId": "ALBJ4LuabUEaYrsX0nGPJeSctN6F63AtR0hdRPvpxmfRbb-WAOCwJSZCVsrbhM9vhrae1inRqYa1C8qUQGDrTZQ",
  "suggestionsViewMode": "SUGGESTIONS_INLINE",
  "documentId": "1PhFCxwa-lf0dqY5e01ib5kQHaB3JgwxY0VMwYKPMisw"
}

