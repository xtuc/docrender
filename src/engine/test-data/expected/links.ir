(
    content: [
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("Sven Sauleau", (
                    link: Some("mailto:Sven.sauleau@gmail.com"),
                )),
            ], false),
        ]),
        Clear,
        Paragraph([
            InlineText([
                TextWithOptions("cloudflare.com", (
                    link: Some("http://cloudflare.com"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text("."),
            ], true),
        ]),
        Clear,
    ],
    styles: {
        "pbNORMAL_TEXT": [
            ("line-height", "1.7249999"),
        ],
    },
)