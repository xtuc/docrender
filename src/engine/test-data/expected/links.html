<style>
  .pbNORMAL_TEXT {
    line-height: 1.7249999;
  }
</style>
<p><a href="mailto:Sven.sauleau@gmail.com">Sven Sauleau</a></p>
<p>
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="http://cloudflare.com"
    >cloudflare.com</a
  >.
</p>
